geotrek/core/models.py

```diff
@@ -946,6 +946,15 @@ def __str__(self):
 class Trail(MapEntityMixin, Topology, StructureRelated):
     topo_object = models.OneToOneField(Topology, parent_link=True, on_delete=models.CASCADE)
     name = models.CharField(verbose_name=_("Name"), max_length=64)
+
+    category = models.ForeignKey(
+        "TrailCategory",
+        on_delete=models.CASCADE,
```

Comment:
if you delete category, you delete the trail -> use models.PROTECT or models.SET_NULL